mobile/android/android-device-file-system.ts

```diff
@@ -61,4 +65,22 @@ export class AndroidDeviceFileSystem implements Mobile.IDeviceFileSystem {
 			}
 		}).future<void>()();
 	}
+
+	public createFileOnDevice(deviceFilePath: string, fileContent: string): IFuture<void> {
+		return (() => {
+			let hostTmpDir = this.getTempDir();
+			let commandsFileHostPath = path.join(hostTmpDir, "temp.commands.file");
+			this.$fs.writeFile(commandsFileHostPath, fileContent).wait();
+
+			// copy it to the device
+			this.transferFile(commandsFileHostPath, deviceFilePath).wait();
+			this.adb.executeShellCommand(["chmod", "0777", `${deviceFilePath}`]).wait();
```

Comment:
I don't think you need the `${}` or the template string here - maybe pass `deviceFilePath` directly?